yarl/_url.py

```diff
@@ -983,8 +983,11 @@ def update_query(self, *args, **kwargs):
         """Return a new URL with query part updated."""
         s = self._get_str_query(*args, **kwargs)
         new_query = MultiDict(parse_qsl(s, keep_blank_values=True))
-        query = MultiDict(self.query)
-        query.update(new_query)
+        if args and args[0] is None:
+            query = new_query
+        else:
+            query = MultiDict(self.query)
+            query.update(new_query)
```

Comment:
Instead of re-testing `args`, check if `s == ""`. `_get_str_query()` will only ever return` ""` if `None` was passed in. We can make use of this to avoid unnecessary round-tripping the query string to `MultiDict` and back:

```suggestion
        s, query = self._get_str_query(*args, **kwargs), None
        if s:
            new_query = MultiDict(parse_qsl(s, keep_blank_values=True))
            query = MultiDict(self.query)
            query.update(new_query)
```

By setting `query = None` at the start we save on parsing an empty string to an empty `MultiDict()` and back again; note that the last line uses `self._get_str_query(query)` again so defaulting `query` to `None` means that `self._val._replace(query="")` is called (where `self._val` is a [`urllib.parse.SplitResult()` named tuple](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.SplitResult)).